filecoin-proofs/src/api/util.rs

```diff
@@ -100,7 +98,56 @@ pub(crate) fn get_t_aux<Tree: MerkleTreeTrait>(
     Ok(res)
 }
 
+/// Instantiates t_aux from default values for access to labels and tree_d, tree_c, tree_r_last
+/// store configs
+#[cfg(feature = "fixed-rows-to-discard")]
+// Silence Clippy warning in order to have the same return value as without this feature.
+#[allow(clippy::unnecessary_wraps)]
+pub(crate) fn get_t_aux<Tree: MerkleTreeTrait>(
+    cache_path: &Path,
+    sector_bytes: u64,
+) -> Result<TemporaryAux<Tree, DefaultPieceHasher>> {
+    trace!(
+        "Instantiating TemporaryAux from default values with cache_path at {:?}",
+        cache_path
+    );
+    let t_aux_path = cache_path.join(CacheKey::TAux.to_string());
+    if Path::new(&t_aux_path).exists() {
+        log::warn!(
+            "`t_aux` file exists, use that file instead of the default values. t_aux={:?}",
+            &t_aux_path
+        );
+        read_t_aux_file(cache_path)
```

Comment:
Is this what we wanted here?  I thought if it existed with the config option set, it would warn loudly that it exists and is being ignored and then continue with the generated default?

Reply:
That's how I understood you and @DrPeterVanNostrand. If we would continue with the default values (and the values in t_aux were different) then things might fail at later stages. I thought that's what you wanted to prevent.

Reply:
I could see the case for either -- but if we're going with the 'supraseal will never touch t_aux` idea, then I think what I said is consistent.  If it's better to support what's there, I'm ok with that, but it's a different model

Reply:
> That's how I understood you and @DrPeterVanNostrand. If we would continue with the default values (and the values in t_aux were different) then things might fail at later stages. I thought that's what you wanted to prevent.

I'm ok with it -- it's just different than the 'fixed-rows-to-discard requiring the defaults only and ignoring t_aux' model we also talked about.  I don't know yet which has the bigger potential for breakage :thinking:

Reply:
I would prefer the "supraseal will never touch t_aux" idea. But wasn't it then previously doing what you now describe? If I look at https://github.com/filecoin-project/rust-fil-proofs/blob/2ee6eda8e577728a3f8dcf3e732859e1442ec201/filecoin-proofs/src/api/util.rs#L101-L106 (that's the old version), there is a warning that the file exists, but then it keeps on going with the defaults.

What I would prefer (but I can see the case of not doing that), would be to error in case the file exists, to indicate that you should enable this feature only on new sectors.

Reply:
Ok, I was confused on the old version then -- I thought it would error if `t_aux` existed, which is the worst option since it always exists for current sectors.  I think I said I'd prefer it to warn loudly that it was being ignored and continue with the defaults in that case.

That said, I can now see the case for warning, but then using the existing (i.e. as it is now).  It's a fine point, but we need (documented) clarity either way.  Maybe I'm leaning toward where it is now (using existing t_aux if found) because it has a better chance of working on installs where a non-standard rows_to_discard was used on the existing sealed sectors (and going forward with supraseal, it would start using the default) :thinking: